src/discordcr/websocket.cr

```diff
@@ -71,6 +82,25 @@ module Discord
       end
     end
 
+    def on_compressed_stream(&handler : Packet ->)
+      @websocket.on_binary do |binary|
+        @zlib_io.write binary
+        next if binary.size < 4 || binary[binary.size - 4, 4] != ZLIB_SUFFIX
+        @zlib_io.rewind
+
+        zlib_reader = (@zlib_reader ||= Zlib::Reader.new(@zlib_io))
+
+        read_size = zlib_reader.read(@buffer_memory)
```

Comment:
I think this `read` call should be looped so the whole packet is read, in case a packet >10MB is received. (Seems unlikely but who knows what Discord might do) This would also allow the buffer size to be configurable in case people want to save some more memory

Reply:
I'll look into this; and yeah I just pulled 10MB from b1nzy's disco lib and figured it was a safe bet.

Do you actually want me to expose some interface to make this configurable? If so I'm not sure off the top of my head the best place to do this, other than to pass it down from the client initializer..

Reply:
Yeah I think passing it down from the client initializer is the best bet, I agree it's a bit clunky but I don't see a better way either